### PR TITLE
feat(crypto)!: remove unnecessary `Sha256` wrapper

### PR DESCRIPTION
**Pull request by melekes**
We are essentially removing a wrapper that took the [32]byte value produced by the standard library and transforming it into a []byte slice.

---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] ~~Updated relevant documentation (`docs/` or `spec/`) and code comments~~
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by melekes** [comment]
@andynog fixed it, thanks!

**Review by ValarDragon — APPROVED** [review]

File: crypto/xsalsa20symmetric/symmetric_test.go [hunk 1]
```diff
@@ -27,7 +26,9 @@ func TestSimpleWithKDF(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-	secret = crypto.Sha256(secret)
+	sum := sha256.Sum256(secret)
+	copy(secret, sum[:])
```
Comment (cason) [inline crypto/xsalsa20symmetric/symmetric_test.go:30]: Just for curiosity, if you don't copy here, the test fails?

File: crypto/doc.go [hunk 2]
```diff
@@ -1,42 +1,29 @@
 // crypto is a customized/convenience cryptography package for supporting
```
Comment (cason) [inline crypto/doc.go:1]:
```suggestion
// crypto is a customized/convenience cryptography package for CometBFT.
```

File: crypto/doc.go [hunk 3]
```diff
@@ -1,42 +1,29 @@
...
 // CometBFT.
```
Comment (cason) [inline crypto/doc.go:2]:
```suggestion

```

File: crypto/doc.go
Comment (cason) [inline crypto/doc.go:1]: We could maybe also document the `PubKey` interface here.

File: crypto/hash.go
Comment (cason) [inline crypto/hash.go:1]: Essentially, this wrapper does transformation of array in slice?

**Review by cason — APPROVED** [review]
Good.

Maybe just to document in the PR/final commit that we are essentially removing a wrapper that took the `[32]byte` value produced by the standard library and transforming it into a `[]byte` slice.

File: crypto/hash.go
Comment (melekes) [reply crypto/hash.go:1]: Yes

**Pull request merged**